controllers/contentlibrary/v1alpha2/contentlibraryitem/contentlibraryitem_controller_unit_test.go

```diff
@@ -0,0 +1,303 @@
+
+				JustBeforeEach(func() {
+					vmi := &vmopv1.VirtualMachineImage{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      clItemCtx.ImageObjName,
+							Namespace: clItem.Namespace,
+						},
+						Spec: vmopv1.VirtualMachineImageSpec{
+							ProviderRef: common.LocalObjectRef{
+								Name: "bogus",
+							},
+						},
+						Status: vmopv1.VirtualMachineImageStatus{
+							ProviderContentVersion: "stale",
+							Firmware:               "should-be-updated",
+						},
+					}
+					Expect(ctx.Client.Create(ctx, vmi)).To(Succeed())
+				})
+
+				It("should update the existing VirtualMachineImage with ContentLibraryItem", func() {
+					clItemCtx.CLItem.Status.ContentVersion += "-updated"
+					Expect(reconciler.ReconcileNormal(clItemCtx)).To(Succeed())
+
+					vmi := getVMI(ctx, clItemCtx)
+					assertVMImageFromCLItem(vmi, clItemCtx.CLItem)
+					Expect(vmi.Status.Firmware).To(Equal(firmwareValue))
+				})
+			})
+
+			When("VirtualMachineImage resource is created and already up-to-date", func() {
+
+				JustBeforeEach(func() {
+					vmi := &vmopv1.VirtualMachineImage{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      clItemCtx.ImageObjName,
+							Namespace: clItemCtx.CLItem.Namespace,
+						},
+						Status: vmopv1.VirtualMachineImageStatus{
+							ProviderContentVersion: clItemCtx.CLItem.Status.ContentVersion,
+							Firmware:               "should-not-be-updated",
+						},
+					}
+					Expect(ctx.Client.Create(ctx, vmi)).To(Succeed())
+				})
+
+				It("should skip updating the VirtualMachineImage with library item", func() {
+					fakeVMProvider.SyncVirtualMachineImageFn = func(_ goctx.Context, _, _ client.Object) error {
+						// Should not be called since the content versions match.
+						return fmt.Errorf("sync-error")
+					}
+
+					Expect(reconciler.ReconcileNormal(clItemCtx)).To(Succeed())
+
+					vmi := getVMI(ctx, clItemCtx)
+					Expect(vmi.Status.Firmware).To(Equal("should-not-be-updated"))
+				})
+			})
+		})
+	})
+
+	Context("ReconcileDelete", func() {
+
+		It("should remove the finalizer from ContentLibraryItem resource", func() {
+			Expect(clItem.Finalizers).To(ContainElement(utils.ContentLibraryItemVmopFinalizer))
+
+			Expect(reconciler.ReconcileDelete(clItemCtx)).To(Succeed())
+			Expect(clItem.Finalizers).ToNot(ContainElement(utils.ContentLibraryItemVmopFinalizer))
+		})
+	})
+}
+
+func getVMI(
+	ctx *builder.UnitTestContextForController,
+	clItemCtx *context.ContentLibraryItemContextA2) *vmopv1.VirtualMachineImage {
+
+	vmi := &vmopv1.VirtualMachineImage{}
+	Expect(ctx.Client.Get(ctx, client.ObjectKey{Name: clItemCtx.ImageObjName, Namespace: clItemCtx.CLItem.Namespace}, vmi)).To(Succeed())
+	return vmi
+}
+
+func assertVMImageFromCLItem(
+	vmi *vmopv1.VirtualMachineImage,
+	clItem *imgregv1a1.ContentLibraryItem) {
+
+	Expect(metav1.IsControlledBy(vmi, clItem)).To(BeTrue())
+	for k := range utils.FilterServicesTypeLabels(clItem.Labels) {
+		Expect(vmi.Labels).To(HaveKey(k))
+	}
```

Comment:
I don't think this is needed here, as we don't add labels for namespaced scope images.